pkg/api/routes/components_test.go

```diff
@@ -67,6 +79,30 @@ func getComponentsHandler(t *testing.T, comps *fleetdb.ServerComponentSlice, cod
 	}
 }
 
+func getComponentTypesHandler(t *testing.T, comps *fleetdb.ServerComponentSlice, code int) http.HandlerFunc {
+	return func(w http.ResponseWriter, _ *http.Request) {
+		var byt []byte
+		if comps != nil {
+			var err error
+			srvResponse := fleetdb.ServerResponse{
+				Records: validComponentTypes,
+			}
+			byt, err = json.Marshal(srvResponse)
+			if err != nil {
+				t.Fatalf("serializing server response: %s", err.Error())
+			}
+		}
+
+		w.WriteHeader(code)
+		if byt != nil {
+			_, err := w.Write(byt)
+			if err != nil {
+				t.Fatalf("writing http response: %s", err.Error())
+			}
+		}
+	}
+}
+
 func TestFetchServerComponents(t *testing.T) {
```

Comment:
I think going to a compile-time list instead of a runtime one will make this a _lot_ simpler.

Reply:
Hi Vince,
Can you please help provide more details about it? I am not sure what I need to do here.
By the way, I have verified the latest version of this PR in sandbox(`alloy` -> `CIS` -> `fleetdb`)